### ISSUE TITLE
How to add page number in footer in PDF generating file

### ISSUE THREAD

**Issue by BhautikAjmera007** [barryvdh/laravel-snappy#146]

**Comment by pflirae** [comment]
I solved this by appending setOption('footer-right','[page]') to the code line. You can place the page number where you like it, just change the 'fotter-right' for other position. The important point is to add the [page] to the option, so the page number is printed.

**Comment by pabloc117** [comment]
By any chance do you know how to set the footer to something like "Page 1 from 2" or something like that?

**Comment by pflirae** [comment]
Yes, just do setOption("footer-right", "Page [page] from [topage]"), where [topage] is the total number of pages of the document.

**Comment by prllxtchz** [comment]
* [page]       Replaced by the number of the pages currently being printed
   * [frompage]   Replaced by the number of the first page to be printed
   * [topage]     Replaced by the number of the last page to be printed
   * [webpage]    Replaced by the URL of the page being printed
   * [section]    Replaced by the name of the current section
   * [subsection] Replaced by the name of the current subsection
   * [date]       Replaced by the current date in system local format
   * [isodate]    Replaced by the current date in ISO 8601 extended format
   * [time]       Replaced by the current time in system local format
   * [title]      Replaced by the title of the of the current page object
   * [doctitle]   Replaced by the title of the output document
   * [sitepage]   Replaced by the number of the page in the current site being converted
   * [sitepages]  Replaced by the number of pages in the current site being converted

**Comment by Trexology** [comment]
can [topage] be used in the blade template?

**Comment by pflirae** [comment]
I think that it not possible to use it on blade, since it is a variable managed by wkhtmtopdf during the conversion to pdf, so it is not visible to blade during the preparation of the view that is rendered to a pdf.

**Comment by Cleitonesc** [comment]
How do I get the page numbers in the footer on the right?
    <script type="text/php">
    if ( isset($pdf) ) {
    
        $font = $fontMetrics->get_font("helvetica", "bold");
        $pdf->page_text(90, 30, "Página: {PAGE_NUM} de {PAGE_COUNT}", $font, 20,  array(0,0,0)); 
    }
</script>

**Comment by kiddtang** [comment]
for those who want to add page number in blade... i found the working answer [here](https://stackoverflow.com/a/13065368/9104189)

```
<html><head><script>
function subst() {
  var vars={};
  var x=document.location.search.substring(1).split('&');
  for(var i in x) {var z=x[i].split('=',2);vars[z[0]] = unescape(z[1]);}
  var x=['frompage','topage','page','webpage','section','subsection','subsubsection'];
  for(var i in x) {
    var y = document.getElementsByClassName(x[i]);
    for(var j=0; j<y.length; ++j) y[j].textContent = vars[x[i]];
  }
}
</script></head><body style="border:0; margin: 0;" onload="subst()">
<table style="border-bottom: 1px solid black; width: 100%">
  <tr>
    <td class="section"></td>
    <td style="text-align:right">
      Page <span class="page"></span> of <span class="topage"></span>
    </td>
  </tr>
</table>
</body></html>
```

so this HTML can be used in the `'footer-html'` option
```
$footer_html = view()->make('_footer')->render();
```
```
$options = [
// other options
    'footer-html'   => $footer_html,
]
```
```
PDF::loadView('pdf', $data)->setOptions($options);  // load your blade with data
```

**Comment by daniel-fodor** [comment]
> 

Hey it is working for you?
